generator/ir_build.go

```
@@ -192,9 +192,16 @@ func buildDomainField(field *protogen.Field, opts *Options) *DomainField {
 
 	if kind == FieldKindMessage {
 		df.MessageTypeName = irMessageNameFromDesc(field.Desc.Message())
+		// Track cross-file references for import generation.
+		if field.Desc.Message().ParentFile() != field.Desc.ParentFile() {
+			df.MessageSourcePath = string(field.Desc.Message().ParentFile().Path())
```

golangci-lint (2.12.2)

[error] unnecessary conversion

(unconvert)